src/mnelab/model.py

```diff
@@ -533,19 +533,24 @@ def get_compatibles(self):
     @data_changed
     def append_data(self, names):
         """Append the given raw data sets."""
-        files = [self.current["data"]]
-        for d in self.data:
+        self.current["name"] += " (appended)"
+        files = []
+        indices = []
+
+        for idx, d in enumerate(self.data):
             if d["name"] in names:
                 files.append(d["data"])
+                indices.append(f"datasets[{idx}]")
+            elif d["name"] == self.current["name"]:
+                files.insert(0, d["data"])
+                indices.insert(0, f"datasets[{idx}]")
```

Comment:
I think it would be easier if you first insert the current dataset, and then in the loop insert the to-be-appended datasets (so everything except for the first). There's not need for the if/elif in the loop.

Reply:
That would work for files[], but the index of the current entry can only be identified correctly in the loop, as it is dependent on whether or not the dataset was duplicated or will be overwritten.

Reply:
But the current dataset is also available as `data`, so the resulting history command is always `mne.concatenate_raws([data, ...])`, or no?

Reply:
That could also work. I will look into it, if it that would result in a cleaner solution.